src/Oscoin/Storage.hs

```diff
@@ -94,7 +94,11 @@ applyTx
 applyTx bs tx = do
     let txHash = Crypto.hash tx
     novel <- isNovelTx bs txHash
-    if | novel     -> Mempool.addTxs [tx] $> Applied [TxAppliedEvent txHash]
+    if | novel     -> do
+            addResult <- Mempool.addTx tx
+            case addResult of
+                Left _  -> pure $ Error []
+                Right _ -> pure $ Applied [TxAppliedEvent txHash]
```

Comment:
It's a fixation, but I would be in favour of writing `Right ()`, so that the minute we will (ever) change the result of the `TxValidator` from `()` to something more structured, the compiler will bark at us.

Reply:
https://github.com/oscoin/oscoin/pull/555/commits/03cd7f8d81783a06a3a24bf3b90ffea3fb75fcb7